api/v1alpha1/argocd_types.go

```diff
@@ -98,7 +98,7 @@ type ArgoCDApplicationControllerSpec struct {
 	// Set this to a duration, e.g. 10m or 600s to control the synchronisation
 	// frequency.
 	// +optional
-	AppSync *metav1.Duration `json:"appSync,omitempty"`
+	AppSync string `json:"appSync,omitempty"`
```

Comment:
ah, yeah you're right it would be, I was maybe too quick to merge this
I wonder about the impact though, if not many people had set that field considering it didn't used to work

The other thing is we do have a couple of other breaking changes coming to the API for v0.8 (removal of `.spec.dex` and `spec.resourceCustomizations`) so maybe we could include this along with them?

cc @iam-veeramalla

Reply:
@jaideepr97 the ideal way would be to convert the `metav1.Duration` to `string` with in the code.  We can consider this as part of the breaking change but there is no prior information to the users about this change. 

If we can revert the commit and handle the behaviour as mentioned above in a new PR, that would be great.

Reply:
@iam-veeramalla sure, I don't mind reverting the commit
an alternate option would be, if we end up implementing a conversion webhook for the other fields we could add the logic you mentioned in there as well

Reply:
Let's not revert the PR, let's fix it forward

Reply:
I did not consider, that this would be a breaking change in the API. A new PR to revert the breaking change is created here: https://github.com/argoproj-labs/argocd-operator/pull/935

Please let me know if anything else comes up 👍

Reply:
Thanks @ItsKev ! Appreciate that you took the time to fix this.